modules/database/src/ioc/rsrv/caservertask.c

```diff
@@ -252,6 +244,21 @@ SOCKET* rsrv_grab_tcp(unsigned short *port)
                 ok = 0;
                 break;
             }
+            if ( listen ( tcpsock, 20 ) < 0 ) {
```

Comment:
I find myself wondering if there is value in having separate error handling for `listen()`.  That is, why not have something like:

> if(bind(tcpsock, &scratch.sa, sizeof(scratch))==0 && listen ( tcpsock, 20 )==0) {